src/utils/pgwire/src/pg_protocol.rs

```diff
@@ -721,6 +740,9 @@ where
             if !is_consume_completed {
                 self.result_cache.insert(portal_name, result_cache);
             }
+
+            // Clear exec context after consuming exec data.
+            session.clear_exec_context();
```

Comment:
There's a `?` in L737. It seems we may fail to clear the context if error occurs.

I would recommend leveraging the RAII approach by making the "set status" as a guard.